core/src/lib/code-generation/code-builder/code-builder-imports.ts

```diff
@@ -0,0 +1,37 @@
+import { IQueueStep } from './queued-code-builder/queue-step';
+import { CodeBuilder } from './code-builder';
+
+type ILibraryImport = {
+    from: string;
+    elements: {
+        element: string;
+        importAs?: string;
+    }[];
+};
+
+export class CodeBuilderImports implements IQueueStep {
+
+    private imports: ILibraryImport[] = [];
+
+    add(element: string, from: string, importAs?: string): void {
```

Comment:
Mein Vorschlag was es meiner Meinung nach leserlicher macht:

add(element: string, from: string, importAs?: string = ''): void {
        let importLib = this.imports.find(x => x.from === from);
        if (!importLib) {
            importLib = { from, elements: [] };
        }
        const libContainsElement = importLib.elements.findIndex(x => x.element === element) !== -1;
        if (libContainsElement) {
            importLib.elements.push({ element, importAs });
        }
            this.imports.push(importLib);
    }